### PR TITLE
Group Endpoints

### PR DESCRIPTION
**Pull request by silvinobarreiros**
- [x] Add `groups` endpoints
- [x] Add new mocks
- [x] Add new router

### CODE REVIEW

File: lib/jumpwire/group.ex [hunk 1]
```diff
@@ -51,39 +51,68 @@ defmodule JumpWire.Group do
 
   def store_policies(res), do: res
 
+  def fetch(org_id, group_id) do
+    case JumpWire.GlobalConfig.get(:groups, {org_id, group_id}) do
+      nil -> {:error, "Group not found"}
+      group -> {:ok, group}
+    end
```
Comment (hexedpackets) [inline lib/jumpwire/group.ex:58]: Why not just use `JumpWire.GlobalConfig.fetch`?

File: lib/jumpwire/group.ex [hunk 2]
```diff
@@ -51,39 +51,68 @@ defmodule JumpWire.Group do
...
+  end
+
+  def put(org_id, group) do
+    JumpWire.GlobalConfig.put(:groups, {org_id, group.id}, group)
+
+    case store_policies({:ok, group}) do
+      {:ok, group} -> {:ok, group}
+      _ -> {:error, "Failed to store policies for group #{group.name}"}
+    end
```
Comment (hexedpackets) [inline lib/jumpwire/group.ex:67]: Passing `{:ok, group}` to `store_policies/1` will always return `{:ok, group}`, the case statement will never be able to match anything else.

File: lib/jumpwire/api/groups_router.ex [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  plug :match
+  plug :put_secret_key_base
+
+  plug Plug.Session,
+    store: :cookie,
+    key: "_jumpwire_key",
+    signing_salt: "I5bC7Dc3"
+
+  plug :fetch_session
+  plug JumpWire.API.AuthPipeline
```
Comment (hexedpackets) [inline lib/jumpwire/api/groups_router.ex:23]: I missed this in the last PR, but the plugs don't need to be duplicated and re-run in every router. All of the plugs in `JumpWire.API.Router` up to `:dispatch` will run before the conn is passed to this module.

File: lib/jumpwire/api/groups_router.ex [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    id = String.downcase(id)
+
+    with {:ok, assertion} <- @sso_module.fetch_active_assertion(conn),
+         {:ok, group} <- JumpWire.Group.fetch(assertion.computed.org_id, id) do
+      send_json_resp(conn, 200, group)
+    else
+      :error ->
+        send_json_resp(conn, 401, %{error: "SSO login required"})
+
+      {:error, message} ->
```
Comment (hexedpackets) [inline lib/jumpwire/api/groups_router.ex:75]:
```suggestion
      _ ->
```

File: lib/jumpwire/group.ex (hunk 1 above)
Comment (silvinobarreiros) [reply lib/jumpwire/group.ex:58]: Could question... updated!

File: lib/jumpwire/group.ex (hunk 2 above)
Comment (silvinobarreiros) [reply lib/jumpwire/group.ex:67]: It doesn't look like `put` will ever return an error unless the policy is not a policy and in that case, an exception happens. That being said, I just went ahead and removed it.

**Review by hexedpackets — APPROVED** [review]
There are a couple of lint warnings worth fixing, otherwise LGTM

**Pull request merged**